### PR TITLE
Change `inputs` type to `AnyObserver<Input>`

### PR DESCRIPTION
**Pull request by mosamer**
Re-opening #127, not sure what was blocking it 🤔  so rebased and created a new PR for review

This PR is implementing migration of inputs property from SubjectType ~> ObserverType as discussed on #120

It includes the following items;

- Change inputs type from InputSubject<Input> ~> AnyObserver<Input>
- Drop InputSubject
- Improved test cases for action inputs behavior

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -15,11 +15,13 @@ Current master
       - `enabled` ~> `isEnabled`
       - `executing` ~> `isExecuting`
    - Deprecate renamed properties
+- Change `inputs` type to `AnyObserver<Input>`
```
Comment (freak4pc) [inline Changelog.md]: I'd write: 

```suggestion
- Remove `InputSubject` in favor of `AnyObserver`
```

**Comment by bobgodwinx** [comment]
This is awesome and `AnyObserver` does the job perfectly. Thanks @mosamer

**Comment by freak4pc** [comment]
Hey! Just realized this was merged to a branch called 4.0 :O 
It doesn't exist on master. Can we fix this?

**Comment by ashfurrow** [comment]
@freak4pc I don't think we can change the target branch after we merge – is this okay as-is?

**Comment by freak4pc** [comment]
@ashfurrow Heya, I already cherry picked this: https://github.com/RxSwiftCommunity/Action/commit/5e7fb77e86e99a45442ad531f0a0191b194777df

:-)

**Comment by mosamer** [comment]
Hi @freak4pc, I am not sure what are we trying to fix 🤔 This issue is part of [Release-4.0 milestone](https://github.com/RxSwiftCommunity/Action/milestone/3) Why do you need to pick it to `master`

**Comment by freak4pc** [comment]
I'm super confused - you've merged 3 PRs of yours lately. This was the only one aimed at a 4.0.0 branch. Everything else was part of `master`. I think it might be better to have a `develop` branch if you want to have a staging thing because otherwise it makes it super confusing.

**Comment by mosamer** [comment]
Sorry, I guess this confusion is a result of this 4.0 milestone being abandoned for a while. branch `4.0` is/was intended as a develop branch for 4.0 issues which are mainly major breaking changes, I am trying to get this done over the next few weeks hopefully

**Comment by freak4pc** [comment]
@mosamer We've already released 4.0.0 to provide support for RxSwift 5, so if you continue on that path we should: 

1. Maybe cut a new (`develop`) branch from `masster`
2. Apply your changes over that
3. We can apply these changes as a minor 4.x release

**Comment by mosamer** [comment]
@freak4pc I am fine with 1 & 2. As for nr. 3, given these are breaking changes to public API are we fine releasing them as a minor 4.x release. wdyt? 🤔

**Comment by freak4pc** [comment]
Uhm I see. That's a tad tricky so I'm not 100% sure. We could bump to 5 but if that's the case maybe we should gather some more changes? Since 4 was just released now.

**Pull request merged**